File: 2271-rearrange-array-elements-by-sign/rearrange-array-elements-by-sign.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var rearrangeArray = function (arr) {
    let pos = []
    let neg = []
    let n = arr.length

    for (let i = 0; i < n; i++) {
        if (arr[i] < 0) {
            neg.push(arr[i])
        } else {
            pos.push(arr[i])
        }
    }

    for (let i = 0; i < n / 2; i++) {
        arr[2 * i] = pos[i];
        arr[2 * i + 1] = neg[i]
    }

    return arr;
};

File: 2271-rearrange-array-elements-by-sign/rearrange-array-elements-by-sign.java
class Solution {
    public int[] rearrangeArray(int[] nums) {
        List<Integer> pos = new ArrayList<>();
        List<Integer> neg = new ArrayList<>();

        for (int num : nums) {
            if (num < 0) {
                neg.add(num);
            } else {
                pos.add(num);
            }
        }

        for (int i = 0; i < nums.length / 2; i++) {
            nums[i * 2] = pos.get(i);
            nums[i * 2 + 1] = neg.get(i);
        }

        return nums;
    }
}